src/Storages/ObjectStorage/DataLakes/DeltaLake/TableSnapshot.cpp

```diff
@@ -225,10 +227,30 @@ class TableSnapshot::Iterator final : public DB::IObjectIterator
         DB::ObjectInfoWithPartitionColumns::PartitionColumnsInfo partitions_info;
         for (const auto & partition_column : context->partition_columns)
         {
-            std::string * value = static_cast<std::string *>(ffi::get_from_string_map(
-                partition_map,
-                KernelUtils::toDeltaString(partition_column),
-                KernelUtils::allocateString));
+            std::string * value;
+            if (context->physical_names_map.empty())
```

Comment:
Add a comment what does it mean. Why we may have physical_names_map empty? Maybe because of columns naming mode for delta lake?